packages/core/src/Stores/client-store.js

```diff
@@ -1140,8 +1146,26 @@ export default class ClientStore extends BaseStore {
     }
 
     @action.bound
-    setDeviceData(device_data) {
-        this.device_data = device_data;
+    setDeviceData() {
+        // Set client URL params on init
+        const url_params = new URLSearchParams(window.location.search);
+        const device_data = {
+            ...(url_params.get('affiliate_token') && { affiliate_token: url_params.get('affiliate_token') }),
```

Comment:
Would it make sense to use the nullish coalescing operator for all of these so we can maintain 1 object with no spread operators and logical ANDs for readability?

Reply:
It appears that this would later overwrite existing device data if we set a value for the property here. Hence the key must be omitted entirely, thus nullish coalescing will not work well here.

Reply:
we want to maintain the previous device_data, if we use nullish coalescing here, it will overwrite the previous data since we put the object property inside the object